File: src/Web/src/model/Door/DoorDto.ts
import { DoorType } from "../../enum/DoorType";
import { Vendor } from "../../enum/Vendor";
import { SensorDto } from "../Sensor/SensorDto";
import { BuzzerDto } from "./BuzzerDto";
import { ReaderDto } from "./ReaderDto";
import { RexDto } from "./RexDto";

export interface DoorDto {
  guid: string;
  name: string;
  vendor: Vendor;
  type: DoorType;
  metadata: string;
  readers: ReaderDto[];
  buzzer: BuzzerDto | null;
  rex: RexDto | null;
  sensor: SensorDto | null;
  locationGuid: string;
  locationName: string;
  isActive: boolean;
  isDefault: boolean;
}

File: src/Modules/Contracts/Core.Contract/DTOs/Door/DoorDto.cs
using SharedKernel.Enums;

namespace Core.Contract.DTOs.Door;

public sealed record DoorDto(
  Guid Guid,
  string Name,
  Vendor Vendor,
  DoorType Type,
  string Metadata,
  List<ReaderDto> Readers,
  BuzzerDto? Buzzer,
  RexDto? RexDto,
  SensorDto? Sensor,
  RelayDto? Relay,
  Guid LocationGuid,
  string LocationName,
  bool IsActive,
  bool IsDefault
);